include/openthread/dns.h

```diff
@@ -57,6 +57,45 @@ extern "C" {
 #define OT_DNS_DEFAULT_SERVER_IP "2001:4860:4860::8888" ///< Defines default DNS Server address - Google DNS.
 #define OT_DNS_DEFAULT_SERVER_PORT 53                   ///< Defines default DNS Server port.
 
+/**
+ * Initializer for otDnsTxtIterator.
+ */
+#define OT_DNS_TXT_ITERATOR_INIT 0
+
+typedef uint16_t otDnsTxtIterator; ///< Used to iterate through the TXT entries.
+
+/**
+ * This structure represents a TXT record entry representing a key/value pair (RFC 6763 - section 6.3).
+ *
+ * The strings buffers pointed to by `mKey` and `mValue` MUST persist and remain unchanged after an instance of such a
+ * structure is passed to OpenThread (as part of `otSrpClientService` instance).
+ *
+ * An array of `otDnsTxtEntry` entries is used in `otSrpClientService` to specify the full TXT record (a list of
+ * entries).
+ *
+ */
+typedef struct otDnsTxtEntry
+{
+    /**
+     * The TXT record key string.
+     *
+     * If `mKey` is not NULL, then the entry is treated as key/value pair with `mValue` buffer providing the value.
+     *   - The entry is encoded as follows:
+     *        - A single string length byte followed by "key=value" format (without the quotation marks).
+              - In this case, the overall encoded length must be 255 bytes or less.
+     *   - If `mValue` is NULL, then key is treated as a boolean attribute and encoded as "key" (with no `=`).
+     *   - If `mValue` is not NULL but `mValueLength` is zero, then it is treated as empty value and encoded as "key=".
+     *
+     * If `mKey` is NULL, then `mValue` buffer is treated as an already encoded TXT-DATA and is appended as is in the
+     * DNS message.
+     *
+     */
+    const char *   mKey;         ///< The TXT record key. Doesn't need to be NULL-terminated string.
+    const uint8_t *mValue;       ///< The TXT record value or already encoded TXT-DATA (depending on `mKey`).
+    uint16_t       mValueLength; ///< Number of bytes in `mValue` buffer.
+    uint8_t mKeyLength; ///< Number of bytes in `mKey` buffer. MUST be set even if `mKey` is a NULL-terminated string.
```

Comment:
align?

Reply:
`clang-format` did this :), I think it is because the comment is too long...